app/build.gradle

```diff
@@ -43,18 +43,17 @@ android {
     }
 
     compileOptions {
-        sourceCompatibility = JavaVersion.VERSION_17
-        targetCompatibility = JavaVersion.VERSION_17
+        sourceCompatibility = JavaVersion.VERSION_11
```

Comment:
I tested that you could even do 1.8, but it is deprecated to compile for java 8 with latest java versions 🤷 

```
Java compiler version 21 has deprecated support for compiling with source/target version 8.
Try one of the following options:
    1. [Recommended] Use Java toolchain with a lower language version
    2. Set a higher source/target version
    3. Use a lower version of the JDK running the build (if you're not using Java toolchain)
For more details on how to configure these settings, see https://developer.android.com/build/jdks.
To suppress this warning, set android.javaCompile.suppressSourceTargetDeprecationWarning=true in gradle.properties.
warning: [options] source value 8 is obsolete and will be removed in a future release
warning: [options] target value 8 is obsolete and will be removed in a future release
warning: [options] To suppress warnings about obsolete options, use -Xlint:-options.
```

Targeting java 11 is still a behavior change but will work for my project since, but from what I've seen a lot of projects still target java 1.8 🤷

Reply:
Done ✅ 
Tested it compiles fine on both Java 21 and 17. `javap -v` confirms it targets Java 8:
<img width="213" alt="image" src="https://github.com/googlemaps/android-maps-ktx/assets/36954793/2459f077-1d8c-48fb-a4f4-0d9ee20bc760">

I update PR title to reflect recent cahnges